packages/@o3r/components/builders/component-extractor/helpers/component/component.extractor.ts

```diff
@@ -263,20 +264,44 @@ export class ComponentExtractor {
     if (!options.exposedComponentSupport) {
       supportedTypes.delete('EXPOSED_COMPONENT');
     }
-    return configs.filter((config) => {
+    return configs.reduce((acc: ComponentConfigOutput[], config) => {
       if (!supportedTypes.has(config.type)) {
         this.logger.warn(`Config type "${config.type}" is not supported for ${config.library}#${config.name}. Excluding it`);
-        return false;
+        return acc;
       }
 
-      const propertiesWithoutDefaultValue = config.properties.filter((property) => property.values === undefined && property.value === undefined);
+      const { propertiesWithDefaultValue, propertiesWithoutDefaultValue } = config.properties.reduce((properties: {
+        propertiesWithDefaultValue: ConfigProperty[];
+        propertiesWithoutDefaultValue: ConfigProperty[];
+      }, property) => {
+        if (property.values === undefined && property.value === undefined) {
```

Comment:
I assume that `undefined` cannot be a legitimate default value ?

Reply:
Yes configuration value cannot be `undefined`